src/main/java/com/microsoft/graph/generated/applications/item/restore/RestoreRequestBuilder.java

```diff
@@ -36,47 +36,54 @@ public RestoreRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @j
+     * @param body The request body
      * @return a {@link DirectoryObject}
      * @throws ODataError When receiving a 4XX or 5XX status code
      * @see <a href="https://learn.microsoft.com/graph/api/directory-deleteditems-restore?view=graph-rest-1.0">Find more info here</a>
      */
     @jakarta.annotation.Nullable
-    public DirectoryObject post() {
-        return post(null);
+    public DirectoryObject post(@jakarta.annotation.Nonnull final RestorePostRequestBody body) {
+        return post(body, null);
     }
     /**
-     * Restore a recently deleted application, group, servicePrincipal, administrative unit, or user object from deleted items. If an item was accidentally deleted, you can fully restore the item. However, security groups cannot be restored. Also, restoring an application doesn't restore the associated service principal automatically. You must call this API to explicitly restore the deleted service principal. A recently deleted item remains available for up to 30 days. After 30 days, the item is permanently deleted.
+     * Restore a recently deleted application, group, servicePrincipal, administrative unit, or user object from deleted items. If an item was accidentally deleted, you can fully restore the item. However, security groups can't be restored. Also, restoring an application doesn't restore the associated service principal automatically. You must call this API to explicitly restore the deleted service principal. A recently deleted item remains available for up to 30 days. After 30 days, the item is permanently deleted.
+     * @param body The request body
      * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
      * @return a {@link DirectoryObject}
      * @throws ODataError When receiving a 4XX or 5XX status code
      * @see <a href="https://learn.microsoft.com/graph/api/directory-deleteditems-restore?view=graph-rest-1.0">Find more info here</a>
      */
     @jakarta.annotation.Nullable
-    public DirectoryObject post(@jakarta.annotation.Nullable final java.util.function.Consumer<PostRequestConfiguration> requestConfiguration) {
-        final RequestInformation requestInfo = toPostRequestInformation(requestConfiguration);
+    public DirectoryObject post(@jakarta.annotation.Nonnull final RestorePostRequestBody body, @jakarta.annotation.Nullable final java.util.function.Consumer<PostRequestConfiguration> requestConfiguration) {
```

Comment:
Flagged breaking changes seem to be a bug fix to allow posting a body.

cc; @baywet @andrueastman

Reply:
first question: how did you spot it? was the new workflow any help?

then, [from the docs](https://learn.microsoft.com/graph/api/directory-deleteditems-restore?view=graph-rest-1.0), it seems the service supports BOTH sending or not sending a body.
our XSLT doesn't have any modification for this, and the parameter is optional

```xml
<Action Name="restore" IsBound="true">
        <Parameter Name="bindingParameter" Type="graph.directoryObject" Nullable="false" />
        <Parameter Name="autoReconcileProxyConflict" Type="Edm.Boolean">
          <Annotation Term="Org.OData.Core.V1.OptionalParameter" />
        </Parameter>
        <ReturnType Type="graph.directoryObject" />
      </Action>
```

The workload [added the new optional parameter to existing actions](https://dev.azure.com/msazure/One/_git/AD-AggregatorService-Workloads/commit/693d356f0ca43800656c39ddf87e0a460beed382?path=/Workloads/Microsoft.DirectoryServices/override/schema-Prod-v1.0.csdl&version=GBmaster&line=2550&lineEnd=2551&lineStartColumn=1&lineEndColumn=1&type=2&lineStyle=plain&_a=files). (default for nullable is true when not specified)

This is what OAS we have for this operation since the change:

```yaml
  '/directory/deletedItems/{directoryObject-id}/restore':
    post:
      requestBody:
        content:
          application/json:
            schema:
              type: object
              properties:
                autoReconcileProxyConflict:
                  type: boolean
                  default: false
                  nullable: true
        required: true
```

And we're missing information here from a strict OpenAPI perspective.

It should instead be

```diff
  '/directory/deletedItems/{directoryObject-id}/restore':
    post:
      requestBody:
        content:
          application/json:
            schema:
              type: object
              properties:
                autoReconcileProxyConflict:
                  type: boolean
                  default: false
                  nullable: true
        required: true
+          */*:
+            schema:
+              type: null
```

But even if we fixed that in yoko, kiota would probably ignore this second entry as is. Even if we fixed that, we'd then need to add overloads in the code dom, and make sure they translate well in every language. (e.g. in typescript, it wouldn't be an overload, but an optional parameter instead).

These are a big set of changes that need to be coordinated across the board. And I'd like to get @darrelmiller to validate my analysis here first.

Reply:
type: null is not allowed in OpenAPI 3.0, so that isn't an option.  https://spec.openapis.org/oas/v3.0.3.html#data-types

Maybe I am missing something here, but isn't the solution to set requestBody.required = false when all of the properties of the request payload are optional?

Reply:
right now kiota is not considering this field, but we could vary based on that, and if the body is NOT required, in the context of Java, project an overload without the body parameter.

Reply:
> first question: how did you spot it? was the new workflow any help?

Yes. Related to comment that is also at https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/2671#discussion_r1765012196

Reply:
in CSharp, we could make all the non required bodies optional parameters I guess.
This kind of change would still be source breaking if the caller was passing the request configuration without naming the parameter.

Reply:
> right now kiota is not considering this field, but we could vary based on that, and if the body is NOT required, in the context of Java, project an overload without the body parameter.

Makes sense. PHP is the only language we support that doesn't support overloading.
And yes, the validation workflow failure helped me catch this quickly

Reply:
ok, can one of you start by creating an issue in yoko for the change please?

Reply:
Created this for now https://github.com/microsoft/OpenAPI.NET.OData/issues/582

Thinking about this a bit more though, if a required parameter is added to this specific action for example, wouldn't we still have a similar problem? The request body would now become required making the optional overloads "dissappear". 

From the generation perspective, would it make sense to always generate a requestBody parameter and have the marshalling not send anything in the scenario with an empty payload/object?

Reply:
Wouldn't adding a required parameter to a previously optional body be a breaking change on the API side? And if an API is happy with that, then the SDKs should reflect the breaking change?

Reply:
Exactly, adding a required parameter to an action is a breaking change at the API level. This is why I didn't consider this scenario. 
We'd still have a potential breaking change though in the current scenario if we use optional parameters in the language. I'm not sure there's a solution to that besides not using optimal parameters which comes with duplication etc...

Reply:
from the meeting today: we could/should patch the metadata in the XSLT in the meanwhile so we don't block all the releases. @Ndiritu can you create in an issue in the metadata repo so the work is prioritized please?